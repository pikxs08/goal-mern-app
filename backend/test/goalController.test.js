const request = require("supertest");
const app = require("../server");
const Goal = require("../models/goalModel");
const User = require("../models/userModel");
const mongoose = require("mongoose");

// import generate auth token
const { protect } = require("../middleware/authMiddleware");

beforeAll(async () => {
  await mongoose.connect(process.env.MONGO_URI, {
    useNewUrlParser: true,
    useUnifiedTopology: true,
    useCreateIndex: true,
    useFindAndModify: false,
  });
});

afterAll(async () => {
  await mongoose.connection.close();
});

// frontend/src/App.test.js

// Mock localStorage
const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn(),
};
global.localStorage = localStorageMock;

describe("POST /api/users/login", () => {
  it("Should return user token", async () => {
    // Perform the HTTP GET request to fetch goals
    const res = await request(app).post("/api/users/login").send({
      email: "pieter@gmail.com",
      password: "test123",
    });

    // Assert the response
    expect(res.statusCode).toEqual(201);

    // save snapshot
    expect(res.body).toMatchSnapshot();
  });
});
