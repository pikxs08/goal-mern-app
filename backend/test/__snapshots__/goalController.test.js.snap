// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /api/users/login Should return user token 1`] = `
{
  "_id": "65edd1c0007bb6439ad7197f",
  "email": "pieter@gmail.com",
  "isMentor": false,
  "name": "Pieter Slabbert",
  "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjY1ZWRkMWMwMDA3YmI2NDM5YWQ3MTk3ZiIsImlhdCI6MTcxMDI0MzA2NiwiZXhwIjoxNzEyODM1MDY2fQ.gaxdbzgNk7gYFNW84nwERlyK4D6ta8nc4BIMEKLAa_Q",
}
`;
